utils/set_configs.bash

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+set -eu
+
+# settings comment
+SETTING_COMMENT='# Raspberry Pi Mouse V3 settings'
+
+# OS architecture (32-bit or 64-bit)
+ARCHITECTURE=$(getconf LONG_BIT)
+
+# dtoverlay setting
+DTOVERLAY='dtoverlay=anyspi:spi0-0,dev="microchip,mcp3204",speed=1000000'
+
+# i2c_baudrate-setting
+DTPARAM='dtparam=i2c_baudrate=62500'
+
+# config-file PATH
+CONFIG_FILE='/boot/firmware/config.txt'
+
+# kernel version
+# remove the dot from the version number and pad single-digit minor versions
+KERNEL_VERSION=$(uname -r | cut -d'-' -f1)
+KERNEL_VERSION_INT=$(echo "$KERNEL_VERSION" | awk -F. '{ printf "%d%02d", $1, $2 }')
```

Comment:
KERNEL_VERSION_INT関数を使って、現在の環境のカーネルバージョン（int型）を取得するように変更したほうがシンプルな処理になりそうです。
また、変数名と関数名が同じだと分かりにくいため、どちらかを変更したほうが良いかと思います。

私の環境では`uname -r | cut -d'-' -f1`を実行すると`6.8.0`が出力され、`echo "6.8.0" | awk -F. '{ printf "%d%02d", $1, $2 }'`を実行すると`608`が出力されます（つまり`KERNEL_VERSION_INT`には`608`が格納されます）。これは正しい挙動でしょうか。個人的には`6.8.0`が`680`に変換されたほうが自然な気がします。

Reply:
カーネルバージョンを扱う変数・関数を修正しました。

- 変数"KERNEL_VERSION_INT"と関数"KERNEL_VERSION_INT()"が同名だったため、関数を"GET_KERNEL_VERSION_INT()"にリネーム
- 関数"GET_KERNEL_VERSION_INT() "を現在のカーネルバージョン取得より前に定義し、自身のカーネルバージョンを整数値に変換する際に使用するように変更

Reply:
> 私の環境ではuname -r | cut -d'-' -f1を実行すると6.8.0が出力され、echo "6.8.0" | awk -F. '{ printf "%d%02d", $1, $2 }'を実行すると608が出力されます（つまりKERNEL_VERSION_INTには608が格納されます）。これは正しい挙動でしょうか。個人的には6.8.0が680に変換されたほうが自然な気がします。

こちらは正しい挙動です。ここの処理ではカーネルバージョンを比較するために少し特殊な処理をしています。

まず、パッチバージョンには文字列が含まれる可能性があり、それが不具合を引き起こす原因となり得るため、比較の際にはパッチバージョンを除外しています。そのため、メジャーバージョンとマイナーバージョンのみでバージョンを比較しています。例えば、カーネルが`6.8.0`の場合、パッチバージョンである`0`を除外し、`6.8`で比較を行います。

また、マイナーバージョンが1桁の場合には、`0`埋めを行っています。これは、カーネルのバージョン番号ではマイナーバージョンも独立した数値として扱われ、バージョンが進むごとに増加していくためです。通常の数値比較とは異なり、`6.8`よりも`6.11`の方が新しいバージョンとなります。もしそのまま数値として扱うと、`6.8`が`6.11`よりも大きく評価されてしまいます。これを防ぐために、`8`の前に`0`を挿入して`08`とし、`6.08`として扱います。最終的には整数に変換して`608`のように比較します。

整数でカーネルバージョンを扱う理由は、2点あります。1点目は小数よりも扱いやすくバグを減らすこと、2点目は本来の小数点を含む見た目から敢えて大きく変えることで加工した数値であるという意図を伝えやすくするためです。


これらは説明しないと意図が伝わりづらいため、以下のようにコード中にコメントを追加しました。

```bash
# compare kernel versions as integers for accurate version comparison (excluding minor versions).
GET_KERNEL_VERSION_INT() {
    # 0-padding is used to avoid minor versions being compared by their first digit.
    echo "$1" | awk -F. '{ printf "%d%02d", $1, $2 }'
}

# kernel version
KERNEL_VERSION=$(uname -r | cut -d'-' -f1)
KERNEL_VERSION_INT=$(GET_KERNEL_VERSION_INT "$KERNEL_VERSION")
```

Reply:
理解できました。
ありがとうございます。